src/ui/free_camera.js

```diff
@@ -317,8 +317,8 @@ class FreeCamera {
         return matrix;
     }
 
-    getDistanceToElevation(elevationMeters: number): number {
-        const z0 = elevationMeters === 0 ? 0 : mercatorZfromAltitude(elevationMeters, this.position[1]);
+    getDistanceToElevation(centerElevationMeters: number, centerLat: number): number {
+        const z0 = centerElevationMeters === 0 ? 0 : mercatorZfromAltitude(centerElevationMeters, centerLat);
```

Comment:
This was clearly a bug that a mercator coordinate was used instead of a latitude one. Would it be better to convert the y-coordinate into latitude value inside the function rather than expecting it as a function parameter? Another option would be to change `elevationMeters` to `elevationMerc` or something.

Reply:
exactly, camera.position is in mercator coordinates.